src/model/SplineSegment.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SplineSegment.h"
+
+#include <cmath>
+
+SplineSegment::SplineSegment(const Point& p, const Point& q):
+        firstKnot(p), firstControlPoint(p), secondKnot(q), secondControlPoint(q) {}
+
+SplineSegment::SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q) {
+    firstKnot = p;
+    firstControlPoint = fp;
+    secondControlPoint = sp;
+    secondKnot = q;
+}
+
+void SplineSegment::draw(cairo_t* cr) {
+    cairo_move_to(cr, firstKnot.x, firstKnot.y);
+    cairo_curve_to(cr, firstControlPoint.x, firstControlPoint.y, secondControlPoint.x, secondControlPoint.y,
+                   secondKnot.x, secondKnot.y);
+}
+
+auto SplineSegment::toVectorOfPoints() -> std::vector<Point> {
+    std::vector<Point> pointArray = {};
+    if (isFlatEnough()) {
+        pointArray = {firstKnot, secondKnot};
+    } else {
+        std::pair<SplineSegment, SplineSegment>&& twoSegments = subdivide(0.5);
+        pointArray = std::move(twoSegments.first.toVectorOfPoints());
+        pointArray.pop_back();
+        std::vector<Point>&& secondArray = twoSegments.second.toVectorOfPoints();
+        pointArray.insert(pointArray.end(), std::make_move_iterator(secondArray.begin()),
+                          std::make_move_iterator(secondArray.end()));
+    }
+    return pointArray;
+}
+
+auto SplineSegment::subdivide(float t) -> std::pair<SplineSegment, SplineSegment> {
+
+    Point b0 = SplineSegment::linearInterpolate(firstKnot, firstControlPoint, t);  // Same as evaluating a Bezier
+    Point b1 = SplineSegment::linearInterpolate(firstControlPoint, secondControlPoint, t);
+    Point b2 = SplineSegment::linearInterpolate(secondControlPoint, secondKnot, t);
+
+    Point c0 = SplineSegment::linearInterpolate(b0, b1, t);
+    Point c1 = SplineSegment::linearInterpolate(b1, b2, t);
+
+    Point d0 = SplineSegment::linearInterpolate(c0, c1, t);  // This would be the interpolated point
+
+    SplineSegment firstPart = SplineSegment(firstKnot, b0, c0, d0);    // first point of each step
+    SplineSegment secondPart = SplineSegment(d0, c1, b2, secondKnot);  // last point of each step
+    return std::make_pair(firstPart, secondPart);
+}
+
+auto SplineSegment::linearInterpolate(const Point& p, const Point& q, float t) -> Point {
+    return Point(p.x * (1 - t) + q.x * t, p.y * (1 - t) + q.y * t);
+}
+
+auto SplineSegment::evaluate(float t) -> Point {
```

Comment:
This function is unused and can be deleted.
Also it can be const:
```suggestion
auto SplineSegment::evaluate(float t) const -> Point {
```